railties/lib/rails/generators/actions.rb

```diff
@@ -344,6 +343,13 @@ def with_indentation(&block) # :doc:
         ensure
           @indentation -= 1
         end
+
+        # Append string to a file with a newline if necessary
+        def append_file_with_newline(path, str, options = {})
+          gsub_file path, /\n?\z/, options do |match|
```

Comment:
Can you add a test that does not affect the original when the target file does not end on a new line?
(I want to confirm in the test that this method does not affect the original file.)

Reply:
@y-yagi 

Thank you for pointing it out. 
Just added tests.